rel-eng/custom/custom.py

```diff
@@ -29,3 +30,122 @@ def _setup_sources(self):
             self.sources.append(os.path.join(self.rpmbuild_sourcedir, filename))
 
+
+SPEC_FILE_TEMPLATE = """
+Name:           {}
+Version:        0
+Release:        0
+Summary:        Test
+License:        Apache-2.0
+{}
+
+%description
+test
+
+%prep
+%autosetup
+
+%build
+
+%install
+
+%files
+
+%changelog
+"""
+
+
+class ContainerBuilder(Builder):
+    '''
+    Builder class adding the script modifying the Dockerfile when pushing to IBS
+    '''
+    push_script = "container_push.sh"
+
+    def run(self, options):
+        info_out("Building package [%s]" % (self.build_tag))
+        self.no_cleanup = options.no_cleanup
+
+        # Reset list of artifacts on each call to run().
+        self.artifacts = []
+
+        try:
+            try:
+                self._create_build_dirs()
+                self.tgz()
+                self.copy_sources()
+                self.generate_spec()
+
+                if options.srpm:
+                    self.srpm()
+            except KeyboardInterrupt:
+                print("Interrupted, cleaning up...")
+        finally:
+            self.cleanup()
+
+        return self.artifacts
+
+    def copy_sources(self):
+        '''
+        Copy the container files to build
+        '''
+        self.copy_push(self.push_script)
+
+        gitdir = os.path.join(self.git_root, self.relative_project_dir)
+        for path in os.listdir(gitdir):
+            file_path = os.path.join(gitdir, path)
+            if os.path.isfile(file_path):
+                target_path = os.path.join(self.rpmbuild_sourcedir, path)
+                self.copy_source(file_path, target_path)
+
+    def copy_source(self, file_path, target_path):
+        debug(f"Copying source {file_path} to {target_path}")
+        shutil.copy2(file_path, target_path)
+        self.sources.append(target_path)
+
+    def tgz(self):
+        self.ran_tgz = True
+
+    def generate_spec(self):
+        '''
+        Generate a spec file to build the SRPM with
+        '''
+        debug(f"Generating spec with sources {self.sources}")
+        sources = "\n".join(["SOURCE{}: {}".format(idx, source) for idx, source in enumerate(self.sources)])
+        spec_content = SPEC_FILE_TEMPLATE.format(self.project_name, sources)
+
+        self.spec_file_name = "{}.spec".format(self.project_name)
+        self.spec_file = os.path.join(self.rpmbuild_sourcedir, self.spec_file_name)
+        with open(self.spec_file, "w") as fd:
+            fd.write(spec_content)
+            
+
+    def copy_push(self, name):
+        script_path = os.path.join(self.git_root, "rel-eng", name)
+        target_script = os.path.join(self.rpmbuild_sourcedir, "push.sh")
+        debug(f"Copying {script_path} to {target_script}")
+        shutil.copy2(script_path, target_script)
+        self.sources.append(target_script)
+
+
+
+helm_chart_files = ["values.yaml", "values.schema.json", "charts", "crds", "templates", "LICENSE", "README.md"]
```

Comment:
Should this be a part of the ChartBuilder class rather than the script?

Reply:
Moved